Lazyy/Lazyy/LazyFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace Lazyy
+{
+    /// <summary>
+    /// создает обьекты для работы либо в однопоточном, лио многопоточном режиме
+    /// </summary>
+    public class LazyFactory<T>
```

Comment:
```suggestion
    public static class LazyFactory<T>
```
И параметр-тип ему не нужен, лучше методы генериками сделать. Тогда не надо будет указывать параметр-тип при вызове.